pkg/agent/manager/manager.go

```diff
@@ -145,7 +150,17 @@ func (m *manager) Run(ctx context.Context) error {
 }
 
 func (m *manager) SubscribeToCacheChanges(selectors cache.Selectors) cache.Subscriber {
-	return m.cache.SubscribeToWorkloadUpdates(selectors)
+	subscriber := m.cache.SubscribeToWorkloadUpdates(selectors)
+	backoff := backoff.NewBackoff(m.clk, svidSyncInterval)
+	// block until all svids are cached and subscriber is notified
+	for {
+		if m.cache.Notify(selectors) {
+			return subscriber
+		}
+		m.c.Log.WithField(telemetry.Selectors, selectors).Info("Waiting for SVID to get cached")
+
+		<-m.clk.After(backoff.NextBackOff())
```

Comment:
We need to accept a `context.Context` in this method now so that we can be informed of context cancellations and return from this method. e.g.
```
select {
case <-ctx.Done():
    return nil, ctx.Err()
case <-m.clk.After(backoff.NextBackoff()):
}
```